Classes/Service/SolrIndexService.php

```diff
@@ -0,0 +1,125 @@
+<?php
+declare(strict_types=1);
+namespace JWeiland\ServiceBw2\Service;
+
+/*
+ * This file is part of the TYPO3 CMS project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+use ApacheSolrForTypo3\Solr\IndexQueue\Item;
+use JWeiland\ServiceBw2\Indexer\Indexer;
+
+/**
+ * Class SolrIndexService
+ *
+ * TODO: Clear existing records
+ *
+ * @package JWeiland\ServiceBw2\Service
+ */
+class SolrIndexService
+{
+    /**
+     * @var Indexer
+     */
+    protected $indexer;
+
+    /**
+     * @var array
+     */
+    protected $alreadyIndexed = [];
+
+    /**
+     * injects indexer
+     *
+     * @param Indexer $indexer
+     * @return void
+     */
+    public function injectIndexer(Indexer $indexer)
+    {
+        $this->indexer = $indexer;
+    }
+
+    /**
+     * Index records
+     *
+     * @param array $records
+     * @param string $type
+     * @param int $rootPageUid
+     * @return void
+     */
+    public function indexRecords(array $records, string $type, int $rootPageUid)
+    {
+        foreach ($records as $key => $record) {
+            if ($record && !$this->alreadyIndexed[$record['id']] && $this->indexRecord($record, $type, $rootPageUid)) {
```

Comment:
Are you sure that this will work? Here you access the record by ArrayKey, but in next line you assign it as ArrayValue.